nixos/modules/services/web-apps/lemmy.nix

```diff
@@ -0,0 +1,142 @@
+{ lib, pkgs, config, ... }:
+with lib;
+let
+  cfg = config.services.lemmy;
+  settingsFormat = pkgs.formats.json { };
+in
+{
+  options.services.lemmy = {
+    enable = mkEnableOption "Enable lemmy server";
+
+    config = mkOption {
+      type = types.submodule {
+        freeformType = settingsFormat.type;
+        options.database = {
+          password = mkOption {
+            type = types.str;
+            description = "password to connect to postgres";
+          };
+        };
+        options.hostname = mkOption {
+          type = types.str;
+          default = null;
+          description = ''the domain name of your instance (eg "lemmy.ml")'';
+        };
+        options.port = mkOption {
+          type = types.port;
+          default = 8536;
+          description = "port where lemmy should listen for incoming requests";
+        };
+        # TODO: this is a secret, should not be in nix-store
```

Comment:
I agree. We have a few ways to take care of this, like generating the actual configuration file with secrets during the service startup. From the rest of your code it looks like you likely understand that, but if you don't please don't hesitate to ask for clarification and/or help.

Let's just make sure we don't add any secrets to the nix store.

Reply:
adressed. Happy to take more feedback in case there is.